packages/web3-utils/src/big_decimal.ts

```diff
@@ -0,0 +1,51 @@
+export class BigDecimal {
+	private readonly bigint: bigint;
+	private readonly decimals: number;
+
+	public constructor(value: bigint | string, decimals?: number) {
+		if (typeof value === 'string' && decimals) {
+			throw new Error('Can not specify decimal with string value');
+		}
+
+		if (typeof value === 'string') {
+			try {
+				const [integer, fraction] = String(value).split('.').concat('');
+				this.decimals = decimals ?? fraction.length;
+				this.bigint = BigInt(`${integer}${fraction.padEnd(this.decimals, '0')}`);
+			} catch {
+				throw new Error(`Invalid value given "${value}". Error: not a valid integer.`);
+			}
+		} else {
+			this.bigint = BigInt(value);
```

Comment:
If `value` can only be `bigint | string` then it's already a `bigint` here and doesn't need to be re-casted

Reply:
Unless this is handling the case where a user passes `42n` as a `bigint`?